apps/dashboard/test/integration/batch_connect_test.rb

```diff
@@ -0,0 +1,31 @@
+require 'html_helper'
+require 'test_helper'
+
+# Test the new radio button functionality in the Jupyter Job Form
+# Testing that radio button with value 0 exists
+# Testing that radio button with value 1 exists
+# Testing that radio button label exists
+
+class BatchConnectTest < ActionDispatch::IntegrationTest
+
+  def setup
+    stub_sys_apps
+  end
+
+  test 'batch_connect_session_context_mode_0 exists in' do
+    get new_batch_connect_session_context_url('sys/bc_jupyter')
+    assert_select 'form input[id="batch_connect_session_context_mode_0"]'   
+  end 
+
+
+  test 'batch_connect_session_context_mode_1 exists in' do
+    get new_batch_connect_session_context_url('sys/bc_jupyter')
+    assert_select 'form input[id="batch_connect_session_context_mode_1"]'   
+  end 
+
+  test 'label for batch_connect_session_context_mode exists in' do
+    get new_batch_connect_session_context_url('sys/bc_jupyter')
+    assert_select 'label[for="batch_connect_session_context_mode"]'   
+  end 
```

Comment:
Since labels are the actual feature that we're adding/fixing I think this test needs to be expanded to ensure they are appropriate. 

We can also combine all these assertions into 1 test case. Ie.e, that radio buttons appear correctly.

In any case, I think these assertions need to be added

1. Label for the form element is overridden. Here you're testing that one or more exist. But this feature is specifically for overriding the default. I.e., this config translates to a new label for the entire form group.
```yaml
  mode:
     label: 'The Mode'
```
```ruby
assert_equal 'The Mode', css_select('label[for="batch_connect_session_context_mode"]').text
```
2. That options are labeled correctly.   You're only testing here that 1 or more labels exist.
```ruby
assert_equal 'Jupyter Lab', css_select('label[for="batch_connect_session_context_mode_1"]').text
assert_equal 'Jupyter Notebook', css_select('label[for="batch_connect_session_context_mode_0"]').text
```

Reply:
Great feedback Jeff. Thanks!